schemas/ViewRoom.json

```diff
@@ -0,0 +1,48 @@
+{
+  "type": "object",
+  "description": "Triggered when the user changes rooms.",
+  "properties": {
+    "eventName": {
+      "enum": ["ViewRoom"]
+    },
+    "trigger": {
+      "description": "The reason for the room change if known.",
+      "type": "string",
+      "oneOf": [
+        {"const": "Created", "description": "Room accessed due to being just created."},
+        {"const": "RoomList", "description": "Room accessed via the room list."},
+        {"const": "RoomDirectory", "description": "Room accessed via the public rooms directory."},
+        {"const": "SpaceHierarchy", "description": "Room accessed via the space hierarchy view."},
+        {"const": "Timeline", "description": "Room accessed via a timeline pill or link in another room."},
+        {"const": "Notification", "description": "Room accessed via a push/desktop notification."},
+        {"const": "Tombstone", "description": "Room accessed via a tombstone at the bottom of a predecessor room."},
+        {"const": "Predecessor", "description": "Room accessed via the predecessor link at the top of the upgraded room."},
+        {"const": "Widget", "description": "Room switched due to widget interaction."},
+        {"const": "MessageSearch", "description": "Room switched due to user interacting with a message search result."},
+        {"const": "MessageUser", "description": "Room switched due to user selecting a user to go to a DM with."},
+        {"const": "VerificationRequest", "description": "Room switched due to user interacting with incoming verification request."},
+
+        {"const": "WebNotificationPanel", "description": "Room accessed via Element Web/Desktop's notification panel."},
+        {"const": "WebSlashCommand", "description": "Room accessed via a slash command in Element Web/Desktop like /goto."},
+        {"const": "WebKeyboardShortcut", "description": "Room accessed via an Element Web/Desktop keyboard shortcut like go to next room with unread messages."},
+        {"const": "WebHorizontalBreadcrumbs", "description": "Room accessed via the Element Web/Desktop horizontal breadcrumbs at the top of the room list."},
+        {"const": "WebVerticalBreadcrumbs", "description": "Room accessed via the Element Web/Desktop vertical breadcrumb hover menu."},
```

Comment:
We have vertical breadcrumbs on Android. Also in the future it could be supported on iOS.
So as a general feedback, I would prefix the values with `Web` only when there is absolutely no reason than the event is used by mobile, today or in the future. In case of doubt, rather not using a suffix.

Reply:
Right but it isn't yet supported on iOS, and the vertical breadcrumbs are incredibly different between Web & Mobile, on Web they are a Labs-only experiment and show up on hover.